src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -589,12 +589,13 @@ private static bool CanBindToTheseConstructorParameters(ParameterInfo[] construc
 
             Debug.Assert(dictionary is not null);
 
-            Type dictionaryObjectType = dictionary.GetType();
+            Type? iDictionaryObjectType = FindOpenGenericInterface(typeof(IDictionary<,>), dictionary.GetType());
 
-            MethodInfo tryGetValue = dictionaryObjectType.GetMethod("TryGetValue", BindingFlags.Public | BindingFlags.Instance)!;
+            Debug.Assert(iDictionaryObjectType is not null);
+
+            MethodInfo tryGetValue = iDictionaryObjectType.GetMethod("TryGetValue", DeclaredOnlyLookup)!;
```

Comment:
The original had different `BindingFlags`. Is `DeclaredOnlyLookup` correct in this case?

Reply:
`DeclaredOnlyLookup` is more specific. as now we are using `IDictionary<,>` interface so using `BindingFlags.DeclaredOnly` will make more sense. The result is the same, but I preferred to use what we used to do before.